client/client.go

```diff
@@ -231,53 +270,189 @@ type (
+	PollerInfo = internal.PollerInfo
+
+	// WorkerVersionCapabilities includes a worker's build identifier
+	// and whether it is choosing to use the versioning feature.
+	// It is an optional component of [Client.PollerInfo].
+	// WARNING: Worker versioning is currently experimental
+	WorkerVersionCapabilities = internal.WorkerVersionCapabilities
+
+	// UpdateWorkerVersioningRulesOptions is the input to [Client.UpdateWorkerVersioningRules].
+	// WARNING: Worker versioning is currently experimental
+	UpdateWorkerVersioningRulesOptions = internal.UpdateWorkerVersioningRulesOptions
+
+	// VersioningConflictToken is a conflict token to serialize calls to Client.UpdateWorkerVersioningRules.
+	// An update with an old token fails with `serviceerror.FailedPrecondition`.
+	// The current token can be obtained with [GetWorkerVersioningRules],
+	// or returned by a successful [UpdateWorkerVersioningRules].
+	// WARNING: Worker versioning is currently experimental
+	VersioningConflictToken = internal.VersioningConflictToken
+
+	// VersioningRampByPercentage is a VersionRamp that sends a proportion of the traffic
+	// to the target Build ID.
+	// WARNING: Worker versioning is currently experimental
+	VersioningRampByPercentage = internal.VersioningRampByPercentage
+
+	// VersioningAssignmentRule is a BuildID  assigment rule for a task queue.
+	// Assignment rules only affect new workflows.
+	// WARNING: Worker versioning is currently experimental
+	VersioningAssignmentRule = internal.VersioningAssignmentRule
+
+	// VersioningAssignmentRuleWithTimestamp contains an assignment rule annotated
+	// by the server with its creation time.
+	// WARNING: Worker versioning is currently experimental
+	VersioningAssignmentRuleWithTimestamp = internal.VersioningAssignmentRuleWithTimestamp
+
+	// VersioningAssignmentRule is a BuildID redirect rule for a task queue.
+	// It changes the behavior of currently running workflows and new ones.
+	// WARNING: Worker versioning is currently experimental
+	VersioningRedirectRule = internal.VersioningRedirectRule
+
+	// VersioningRedirectRuleWithTimestamp contains a redirect rule annotated
+	// by the server with its creation time.
+	// WARNING: Worker versioning is currently experimental
+	VersioningRedirectRuleWithTimestamp = internal.VersioningRedirectRuleWithTimestamp
+
+	// VersioningOpInsertAssignmentRule is an operation for UpdateWorkerVersioningRulesOptions
+	// that inserts the rule to the list of assignment rules for this Task Queue.
+	// The rules are evaluated in order, starting from index 0. The first
+	// applicable rule will be applied and the rest will be ignored.
+	// By default, the new rule is inserted at the beginning of the list
+	// (index 0). If the given index is too larger the rule will be
+	// inserted at the end of the list.
+	// WARNING: Worker versioning is currently experimental
+	VersioningOpInsertAssignmentRule = internal.VersioningOpInsertAssignmentRule
```

Comment:
Generally initialisms in Go prefer constant case so either `op` or `OP`. I would prefer `Operation` here since `op` could be `operation` or `option` and without reading the docs I can't tell. @cretz any thoughts?
 
https://google.github.io/styleguide/go/decisions#initialisms

Reply:
I was mostly following what we had with the old versioning, e.g., `BuildIDOpAddNewIDInNewDefaultSet`. I guess that strictly speaking Op is not an initialism, it is just an abbreviation, but we could make it OP. My only concern is that in the middle of a word sometimes it is harder to "parse", i.e.,  VersioningOPAddRedirectRule vs VersioningOpAddRedirectRule , is "OPA" or "PA" (you need look-ahead :-) ).

About the confusion with Options, we always have "Options" with no shortcuts, so that is less confusing with "OP" for operations. I worry if we extend operations it is just too much typing, e.g., VersioningOperationDeleteRedirectRule...

Anyway, happy to change things, @cretz comments?

Reply:
I have no strong opinion. `Operation` works for me. I do appreciate how we're prefixing all of these w/ the same prefix though.